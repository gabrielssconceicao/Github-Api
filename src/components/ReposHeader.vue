<script setup lang="ts">
const props = defineProps<{ name: string }>()
</script>

<template>
  <header class="bg-header p-2 flex items-center text-white">
    <RouterLink to="/" class="w-14 border-white border-2 p-1  flex justify-center items-center">Back
    </RouterLink>
    <p class="flex-1 text-center">
      <span class="font-bold font-mono text-xl text-primary">{{ props.name }}</span>
      Repos
    </p>
  </header>
</template>
