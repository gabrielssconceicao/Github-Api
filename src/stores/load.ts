import { ref } from 'vue';
import { defineStore } from 'pinia';

export const useLoadStore = defineStore('load', () => {
  const loading = ref(false);
  function setLoading() {
    loading.value = true;
  }
  function setNotLoading() {
    loading.value = false;
  }
  return { loading, setLoading, setNotLoading };
});
