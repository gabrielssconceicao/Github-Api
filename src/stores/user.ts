import { ref } from 'vue';
import { defineStore } from 'pinia';
import type { IUserResponse } from '@/types/user';
import { fetchUser } from '@/hooks/fetchUser';
export const useUserStore = defineStore('counter', () => {
  const githubUser = ref<IUserResponse>({} as IUserResponse);

  async function searchUser(userLogin: string) {
    const data = await fetchUser(userLogin);
    githubUser.value = data;
  }

  return { githubUser, searchUser };
});
