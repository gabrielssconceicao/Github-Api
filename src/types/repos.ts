export interface IRepos {
  name: string;
  description: string | null;
  html_url: string;
  language: string;
}
