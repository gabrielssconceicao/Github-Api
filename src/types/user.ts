export interface IUser {
  name: null | string;
  login: string;
  avatar_url: string;
  followers: number;
  following: number;
  location: null | string;
  html_url: string;
  bio: null | string;
  public_repos: number;
  repos_url: string;
}

export interface IUserResponse {
  user: IUser;
  status: number;
}
