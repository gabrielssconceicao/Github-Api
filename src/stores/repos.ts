import { defineStore } from 'pinia';
import { ref, computed } from 'vue';

import { type IRepos } from '@/types/repos';

import { langSeparator } from '@/helpers/langSeparator';
import { useLoadStore } from './load';
export const useReposStore = defineStore('repos', () => {
  const repos = ref<IRepos[]>([]);
  const reposUrl = ref('');
  const filterValue = ref('');

  function setRepos(reposValue: IRepos[]) {
    repos.value = reposValue;
  }

  const setFilter = (value: string) => {
    filterValue.value = value;
  };

  function setReposUrl(url: string) {
    reposUrl.value = url;
  }
  async function getRepos() {
    const useLoad = useLoadStore();
    useLoad.setLoading();
    try {
      const response = await fetch(reposUrl.value);
      const data: IRepos[] = await response.json();
      const dataWithoutNull = data.map((e) => {
        if (e.language) return e;
        return { ...e, language: 'No Language' };
      });
      setRepos(dataWithoutNull);
    } catch (error) {
      setRepos([]);
    } finally {
      useLoad.setNotLoading();
    }
  }

  const filteredRepos = computed(() => {
    return filterValue.value !== ''
      ? repos.value.filter((repo) => repo.language === filterValue.value)
      : repos.value;
  });

  const filteredReposCount = computed(
    () => repos.value && langSeparator(repos.value),
  );

  return {
    filteredRepos,
    getRepos,
    filteredReposCount,
    setFilter,
    setReposUrl,
    setRepos,
    repos,
    reposUrl,
    filterValue,
  };
});
