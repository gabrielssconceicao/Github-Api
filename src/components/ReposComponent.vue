<script setup lang="ts">
import { type IRepos } from '@/types/repos'
import { getLangColor } from '@/helpers/getLangColor'
const props = defineProps<{ repos: IRepos }>()
</script>
<template>
  <div :style="{ '--lang-color': getLangColor(props.repos.language) }" id="repo">
    <h3 class="text-center p-1 font-bold font-serif">{{ props.repos.name }}</h3>
    <p class="text-justify font-mono text-base break-words mt-3 mb-2" v-if="props.repos.description"
      aria-label="repo-description">{{
        props.repos.description }}</p>
    <div class="flex justify-between p-2">
      <p class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
        aria-label="repo-language">{{ props.repos.language
        }}
      </p>
      <a :href="props.repos.html_url" target="_blank"
        class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]">See
        Repo</a>
    </div>
  </div>
</template>
<style scoped>
#repo {
  @apply bg-zinc-600 p-3 relative max-w-[300px] max-h-[150px] rounded-xl shadow-md before:w-2 before:absolute before:bg-[var(--lang-color)] before:top-0 before:left-0 before:bottom-0 before:rounded-tl-xl before:rounded-bl-xl md:text-xl md:w-[500px]
}
</style>
