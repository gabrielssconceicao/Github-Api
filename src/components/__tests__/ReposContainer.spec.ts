import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import ReposContainer from '../RepositoriesContainer.vue';
import ReposHeader from '../ReposHeader.vue';
import ReposFilter from '../ReposFilterComponent.vue';
import ReposComponent from '../ReposComponent.vue';

import { useUserStore } from '@/stores/user';
import { useReposStore } from '@/stores/repos';

describe('<RepositoriesContainer/>', () => {
  it('should render with the correct data', async () => {
    const useUser = useUserStore();
    const useRepos = useReposStore();
    useUser.setInputValue('resolveTest');
    await useUser.searchUser();

    await useRepos.getRepos();

    const wrapper = mount(ReposContainer);
    expect(wrapper.findComponent(ReposHeader).exists()).toBe(true);
    expect(wrapper.find('section#repos').exists()).toBe(true);
    expect(wrapper.findComponent(ReposFilter).exists()).toBe(true);
    expect(wrapper.findComponent(ReposComponent).exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should filter', async () => {
    const useUser = useUserStore();
    const useRepos = useReposStore();
    useUser.setInputValue('resolveTest');
    await useUser.searchUser();

    await useRepos.getRepos();

    const wrapper = mount(ReposContainer);
    const filter = wrapper.findComponent(ReposFilter);
    await filter.findAll('button')[1].trigger('click');

    expect(wrapper.findAll('div#repo').length).toBe(
      useRepos.filteredRepos.length,
    );

    expect(wrapper.element).toMatchSnapshot();
  });
});
