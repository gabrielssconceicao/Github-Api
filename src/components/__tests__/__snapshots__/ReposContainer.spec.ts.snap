// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RepositoriesContainer/> > should filter 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="bg-header p-2 flex items-center text-white"
    data-v-f09d8ef3=""
  >
    <routerlink
      class="w-14 border-white border-2 p-1 flex justify-center items-center"
      to="/"
    >
      Back 
    </routerlink>
    <p
      class="flex-1 text-center"
    >
      <span
        class="font-bold font-mono text-xl text-primary"
      >
        Teste User
      </span>
       Repos 
    </p>
  </header>
  <section
    class="h-[calc(100dvh-51.2px)] overflow-y-auto relative"
    data-v-f09d8ef3=""
    id="repos"
  >
    <div
      class="w-full grid grid-cols-3 gap-2 p-1 sm:grid-cols-5 md:flex md:flex-wrap md:justify-around md:gap-2 bg-section sticky top-0 z-10"
      data-v-b209dafd=""
      data-v-f09d8ef3=""
    >
      <button
        class="clear"
        data-v-b209dafd=""
      >
        <span
          data-v-b209dafd=""
        >
          Clear
        </span>
      </button>
      
      <button
        class="selected"
        data-v-b209dafd=""
        style="--text-color: #ffffff; --bg-color: #1f1f1f;"
        value="Javascript"
      >
        <span
          data-v-b209dafd=""
        >
          Javascript
        </span>
        <span
          data-v-b209dafd=""
        >
          3
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #e34c26; --bg-color: #1f1f1f;"
        value="HTML"
      >
        <span
          data-v-b209dafd=""
        >
          HTML
        </span>
        <span
          data-v-b209dafd=""
        >
          2
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #3572A5; --bg-color: #1f1f1f;"
        value="Python"
      >
        <span
          data-v-b209dafd=""
        >
          Python
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #CC342D; --bg-color: #1f1f1f;"
        value="Ruby"
      >
        <span
          data-v-b209dafd=""
        >
          Ruby
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #b07219; --bg-color: #1f1f1f;"
        value="Java"
      >
        <span
          data-v-b209dafd=""
        >
          Java
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #f34b7d; --bg-color: #1f1f1f;"
        value="C++"
      >
        <span
          data-v-b209dafd=""
        >
          C++
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #ffffff; --bg-color: #1f1f1f;"
        value="Typescript"
      >
        <span
          data-v-b209dafd=""
        >
          Typescript
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      
    </div>
    <div
      class="max-w-7xl mx-auto overflow-y-auto py-8"
      data-v-f09d8ef3=""
    >
      <transition-group-stub
        appear="false"
        class="grid gap-3 px-2 grid-cols-1 justify-items-stretch"
        css="true"
        data-v-f09d8ef3=""
        id="repos-component"
        name="fade"
        persisted="false"
        tag="div"
      >
        
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 1
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 4
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 5
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        
      </transition-group-stub>
    </div>
  </section>
  
</div>
`;

exports[`<RepositoriesContainer/> > should render with the correct data 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="bg-header p-2 flex items-center text-white"
    data-v-f09d8ef3=""
  >
    <routerlink
      class="w-14 border-white border-2 p-1 flex justify-center items-center"
      to="/"
    >
      Back 
    </routerlink>
    <p
      class="flex-1 text-center"
    >
      <span
        class="font-bold font-mono text-xl text-primary"
      >
        Teste User
      </span>
       Repos 
    </p>
  </header>
  <section
    class="h-[calc(100dvh-51.2px)] overflow-y-auto relative"
    data-v-f09d8ef3=""
    id="repos"
  >
    <div
      class="w-full grid grid-cols-3 gap-2 p-1 sm:grid-cols-5 md:flex md:flex-wrap md:justify-around md:gap-2 bg-section sticky top-0 z-10"
      data-v-b209dafd=""
      data-v-f09d8ef3=""
    >
      <button
        class="clear"
        data-v-b209dafd=""
      >
        <span
          data-v-b209dafd=""
        >
          Clear
        </span>
      </button>
      
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #ffffff; --bg-color: #1f1f1f;"
        value="Javascript"
      >
        <span
          data-v-b209dafd=""
        >
          Javascript
        </span>
        <span
          data-v-b209dafd=""
        >
          3
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #e34c26; --bg-color: #1f1f1f;"
        value="HTML"
      >
        <span
          data-v-b209dafd=""
        >
          HTML
        </span>
        <span
          data-v-b209dafd=""
        >
          2
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #3572A5; --bg-color: #1f1f1f;"
        value="Python"
      >
        <span
          data-v-b209dafd=""
        >
          Python
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #CC342D; --bg-color: #1f1f1f;"
        value="Ruby"
      >
        <span
          data-v-b209dafd=""
        >
          Ruby
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #b07219; --bg-color: #1f1f1f;"
        value="Java"
      >
        <span
          data-v-b209dafd=""
        >
          Java
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #f34b7d; --bg-color: #1f1f1f;"
        value="C++"
      >
        <span
          data-v-b209dafd=""
        >
          C++
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      <button
        class=""
        data-v-b209dafd=""
        style="--text-color: #ffffff; --bg-color: #1f1f1f;"
        value="Typescript"
      >
        <span
          data-v-b209dafd=""
        >
          Typescript
        </span>
        <span
          data-v-b209dafd=""
        >
          1
        </span>
      </button>
      
    </div>
    <div
      class="max-w-7xl mx-auto overflow-y-auto py-8"
      data-v-f09d8ef3=""
    >
      <transition-group-stub
        appear="false"
        class="grid gap-3 px-2 grid-cols-1 justify-items-stretch"
        css="true"
        data-v-f09d8ef3=""
        id="repos-component"
        name="fade"
        persisted="false"
        tag="div"
      >
        
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 1
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #e34c26;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 2
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              HTML
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #3572A5;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 3
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Python
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 4
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 5
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Javascript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #e34c26;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 6
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              HTML
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #CC342D;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 7
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Ruby
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #b07219;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 8
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Java
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #f34b7d;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 9
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              C++
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        <div
          data-v-7152ca0f=""
          data-v-f09d8ef3=""
          id="repo"
          style="--lang-color: #ffffff;"
        >
          <h3
            class="text-center p-1 font-bold font-serif break-all"
            data-v-7152ca0f=""
          >
            Teste Repo 10
          </h3>
          <p
            aria-label="repo-description"
            class="text-center font-mono text-base break-words mt-3 mb-2"
            data-v-7152ca0f=""
          >
            lorem ipsum dolor sit amet
          </p>
          <div
            class="flex justify-around p-2"
            data-v-7152ca0f=""
          >
            <p
              aria-label="repo-language"
              class="px-2 py-1 font-sans border-2 border-[var(--lang-color)] text-[var(--lang-color)]"
              data-v-7152ca0f=""
            >
              Typescript
            </p>
            <a
              class="border-white border-2 px-2 py-1 text-white transition-colors hover:border-[var(--lang-color)] hover:text-[var(--lang-color)]"
              data-v-7152ca0f=""
              href="https://github.com/testeuser/testerepo"
              target="_blank"
            >
              See Repo
            </a>
          </div>
        </div>
        
      </transition-group-stub>
    </div>
  </section>
  
</div>
`;
